app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -53,9 +69,33 @@ class MainActivity : AppCompatActivity(), ApplicationContract.View {
     }
 
     override fun showData(data: List<ApplicationContract.TrainJourney>) {
-        println(data)
+        val numberOfJourneys = data.size
+        val departureTimes = mutableListOf<String>()
+        val arrivalTimes = mutableListOf<String>()
+        val stationChanges = mutableListOf<String>()
+        val pricesList = mutableListOf<String>()
+
+        for (i in 0 until (numberOfJourneys-1)){
```

Comment:
I'd probably move this logic to SharedCode so you don't need to duplicate it for Android and iOS :)

Also, you could define a data class for the new information to display and then do something like:
data.map {
   return DisplayData(it.departureTime, it.ArrivalTime, etc..)
}